src/components/side-panel/property-inputs/ConfidenceProperties.tsx

```diff
@@ -0,0 +1,115 @@
+import { useContext, useEffect, useState } from "react";
+import "../../side-panel/side-panel.css";
+import { colors, dashStyles } from "../../../helper-functions/chart-helpers";
+import CustomSelect, { CustomSelectProps } from "../../select/CustomSelect";
+import ChartContext from "../../../context/ChartContext";
+import ColorOption from "../../select/ColorOption";
+
+interface Props {
+  activeSeries: string;
+  columnNames: string[];
+}
+
+const ConfidenceProperties = ({
+  activeSeries,
+  columnNames,
+}: Props): JSX.Element => {
+  const { selectedDimensions, setSelectedDimensions, chartProperties } =
+    useContext(ChartContext);
+  const [prevChartType, setPreviousChartType] = useState<any>("Line");
+  const [intervalType, setIntervalType] = useState("---");
+  const chartType = chartProperties?.chartTypes?.chartType;
+
+  useEffect(() => {
+    if (prevChartType === "Line" && intervalType === "intervals") {
+      updateAllDimensions("intervalType", "---");
+    }
+    setPreviousChartType(chartType);
+  }, [chartType]);
+
+  // Get the selected dimension based on the active series prop
+  const selectedDimension = selectedDimensions.find(
+    (d) => d.name === activeSeries,
+  );
+
+  const updateAllDimensions = (property: string, value: any) => {
```

Comment:
first reaction here - logically `updateAlldimensions` should be a loop of `updateDimension` (or the appropriate js mechanism of applying `update Dimension` to each dimension), its implied from the naming convention, if that's not the case then something probably needs renaming to make the differing behaviour more obvious.

if that's rooted in a different langiage approach then ignore (but tell me why I'm wrong, so I know for next time 😄 )

Reply:
In this instance because of the way the functions work it avoiding needing to repeatedly set the dimensions.
Could possibly introduce an extra function, updateActiveDimension. So both updateActiveDimension and updateAllDimensions would pass their mapped dimensions into updateDimensions. Would something like that work better?

Reply:
if I'm following, `updateDimension` is updating a dimension _only where a user has explicitly requested_ confidence information be displayed?

if that's right I'd just tweak the name to something like `updateDimensionConfidenceDisplay` or somesuch (that's an awful name 😄 ).

Just distinguishing the different behaviour at a glance is all I was after. A name change of that ilk makes it obvious why we just don't loop it in `updateAllDimensions`.

Reply:
Not quite, it updates whenever a new property is selected, only time it wouldn't is if a chart other than line is selected and the user tries to put on intervals, then an alert pops up.

I've changed it to what you originally suggested but the mirror of that. The default is updating all dimensions, unless when only a selected dimension needs updating and a filter is applied.